sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -1542,6 +1549,60 @@ public virtual void RequestAuthorizationException(string identifier, string endp
                 WriteEvent(RequestAuthorizationExceptionEvent, identifier, endpoint, exception);
             }
         }
+
+        [Event(DrainLinkStartEvent, Level = EventLevel.Informational, Message = "{0}: Starting drain operation.")]
+        public void DrainLinkStart(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(DrainLinkStartEvent, identifier);
+            }
+        }
+
+        [Event(DrainLinkCompleteEvent, Level = EventLevel.Informational, Message = "{0}: Drain operation completed.")]
+        public void DrainLinkComplete(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(DrainLinkCompleteEvent, identifier);
+            }
+        }
+
+        [Event(DrainLinkExceptionEvent, Level = EventLevel.Error, Message = "{0}: Drain operation failed with exception: {1}")]
+        public void DrainLinkException(string identifier, string exception)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(DrainLinkExceptionEvent, identifier, exception);
+            }
+        }
+
+        [Event(CloseLinkStartEvent, Level = EventLevel.Informational, Message = "{0}: Starting disposal of AMQP link.")]
+        public void CloseLinkStart(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(CloseLinkStartEvent, identifier);
+            }
+        }
+
+        [Event(CloseLinkCompleteEvent, Level = EventLevel.Informational, Message = "{0}: Disposal of AMQP link completed.")]
+        public void CloseLinkComplete(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(CloseLinkCompleteEvent, identifier);
+            }
+        }
+
+        [Event(CloseLinkExceptionEvent, Level = EventLevel.Error, Message = "{0}: Disposal of AMQP link failed with exception: {1}")]
```

Comment:
We can remove this one.   Looks like we're using `SafeClose` everywhere.   I forgot about that one.